docs/deleting.md

```diff
@@ -214,6 +215,52 @@ Output:
 DELETE FROM "test" RETURNING "test".*
 ```
 
+<a name="seterror"></a>
+**[`SetError`](https://godoc.org/github.com/doug-martin/goqu/#DeleteDataset.SetError)**
+
+Sometimes while building up a query with goqu you will encounter situations where certain
+preconditions are not met or some end-user contraint has been violated. While you could
+track this error case separately, goqu provides a convenient built-in mechanism to set an
+error on a dataset if one has not already been set to simplify query building.
+
+Set an Error on a dataset:
+
+```go
+func GetDelete(name string, value string) *goqu.DeleteDataset {
+
+    var ds = goqu.Delete("test").
+        Where(goqu.C(name).Eq(value))
+
+    if len(name) == 0 {
```

Comment:
I know this is pedantic but I think it makes sense to put the validation before setting the where. In the end it wont make a difference based on the implementation; but setting the error before creating the clause signifies intention.

I often struggle with showing intention vs how its actually implemented, but in this case validation should come first.

If you disagree let me know, but I want docs to demonstrate how it should be used and not necessarily how it could be used.

Reply:
I totally agree. I didn't spend a ton of time thinking about the examples from that perspective but you're totally right. I'll push an update. Let me know if you find other things that are confusing.

Reply:
Thank you for being so receptive to comments! I think the same applies for the rest of the examples (they all seem to be based off of this example). Should be a quick update that will helps many users in the long run.

Reply:
Once you update Ill merge update history and release!

Reply:
Absolutely! Yes, I copy-pasted this example to the others and made edits where necessary.

I'll fix the other examples too.